packages/adapter-components/src/definitions/user/user_config.ts

```diff
@@ -19,24 +19,28 @@ import { createClientConfigType, ClientBaseConfig, ClientRateLimitConfig } from
 import { UserFetchConfig, createUserFetchConfigType } from './fetch_config'
 import { UserDeployConfig, createChangeValidatorConfigType, createUserDeployConfigType } from './deploy_config'
 
-export type UserConfig = {
+export type UserConfig<TCustomNameMappingOptions extends string> = {
```

Comment:
maybe add
```suggestion
export type UserConfig<TCustomNameMappingOptions extends string = never> = {
```
so that we don't have to override it when there are no customizations? (might apply to other types as well)